app/src/pages/CardSubmission/index.tsx

```diff
@@ -38,6 +38,15 @@ function CardSubmission() {
       : CardSubmissionState.Submitting
   )
 
+  React.useEffect(() => {
+    if (
+      currentGame.num_entries_per_player &&
+      numSubmitted < currentGame.num_entries_per_player
+    ) {
+      setCardSubmissionState(CardSubmissionState.Submitting)
+    }
+  }, [numSubmitted])
```

Comment:
I think we need this because we are setting the state (and not deriving it from props each time)

the other way to do this... is on form submit, just wait until the subscription picks up the new cards and then goes to the waiting state (all driven from props).